src/VisualStudio/Core/Def/TableDataSource/DiagnosticTableItem.cs

```diff
@@ -98,24 +98,21 @@ public bool Equals(DiagnosticData left, DiagnosticData right)
                 // location-less or project level diagnostic:
                 if (leftLocation == null ||
                     rightLocation == null ||
-                    leftLocation.OriginalFilePath == null ||
-                    rightLocation.OriginalFilePath == null ||
+                    leftLocation?.OriginalFileSpan?.Path == null ||
+                    rightLocation?.OriginalFileSpan?.Path == null ||
                     left.DocumentId == null ||
                     right.DocumentId == null)
                 {
                     return left.Equals(right);
                 }
 
                 return
-                    leftLocation.OriginalStartLine == rightLocation.OriginalStartLine &&
-                    leftLocation.OriginalStartColumn == rightLocation.OriginalStartColumn &&
-                    leftLocation.OriginalEndLine == rightLocation.OriginalEndLine &&
-                    leftLocation.OriginalEndColumn == rightLocation.OriginalEndColumn &&
+                    leftLocation?.OriginalFileSpan?.Span == rightLocation?.OriginalFileSpan?.Span &&
                     left.Severity == right.Severity &&
                     left.IsSuppressed == right.IsSuppressed &&
                     left.Id == right.Id &&
                     left.Message == right.Message &&
-                    leftLocation.OriginalFilePath == rightLocation.OriginalFilePath;
+                    leftLocation?.OriginalFileSpan?.Path == rightLocation?.OriginalFileSpan?.Path;
```

Comment:
could likely not test Path/Span independently and just do equality on the full FileSpan entity itself.